### PR TITLE
Arihan/mar 50 dashboard usage page

### PR DESCRIPTION
**Pull request by arihanv**
- Added support for create and delete custom instructions
- Usage table
- Logging of usage (prompt + completion tokens)
- Charts displaying usage day aggregated by day

### CODE REVIEW

**Comment by tedspare** [comment]
Should this be merging into `dashboard-init` as opposed to main or staging?

**Comment by arihanv** [comment]
> Should this be merging into `dashboard-init` as opposed to main or staging?

You right

File: app/api/webhook/github/route.ts [hunk 1]
```diff
@@ -299,8 +302,9 @@ Your instructions: ${instructions || 'do nothing'}.
 		await maige({
 			input: prompt,
 			octokit,
-			prisma,
+			// prisma,
```
Comment (tedspare) [inline app/api/webhook/github/route.ts:305]: Should this be uncommented?

File: app/dashboard/(base)/usage/[[...metric]]/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	}
+}) {
+	if (
+		params.metric &&
+		(params.metric.length > 1 ||
+			!paramsSchema.safeParse(params?.metric[0]).success)
+	)
+		return notFound()
+
+	const route = params.metric ? params.metric[0] : ''
```
Comment (tedspare) [inline app/dashboard/(base)/usage/[[...metric]]/layout.tsx:24]: Beautifully simple

File: lib/components/dashboard/usage/charts.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+			<TestChart
+				data={convertedUsage}
+				category='runs'
+				color='green'
+			/>
+		)
+
+	if (route === 'tokens')
+		return (
+			<TestChart
```
Comment (tedspare) [inline lib/components/dashboard/usage/charts.tsx:60]: Rename `TestChart` to `TokenChart`

File: lib/components/dashboard/usage/table-wrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+					name: true,
+					id: true
+				}
+			}
+		}
+	})
+	const end = performance.now()
+	const timeTaken = end - start
+
+	console.log("Rendering Table")
```
Comment (tedspare) [inline lib/components/dashboard/usage/table-wrapper.tsx:82]: Remove console.log

**Comment by tedspare** [comment]
- Co-reviewed over video call yesterday
- Looks great to me. So snappy 😍 
- Will merge and touch up UI in staging

image

**Pull request merged**